controllers/glanceapi_controller.go

```diff
@@ -458,6 +473,11 @@ func (r *GlanceAPIReconciler) reconcileInit(
 
 		svc.AddAnnotation(map[string]string{
 			service.AnnotationEndpointKey: endpointTypeStr,
+			// Add to the current service an annotation that refers to the associated
+			// headless service: this information will be used by the openstack-operator
+			// to have an additional SubjectName that allows to reach every replica with
+			// an https endpoint
+			"headless": headlessSvcName,
```

Comment:
Holding the patch for now because I might need @stuggi's input on this: here I annotate the SVC with this key that will be resolved by the openstack-operator to build an additional `sn` entry in the `CertificateRequest` (I'll come up with a patch today for that part).
Should we need to choose a more generic annotation structure/naming to pass that information?
e.g.

```
tls:
  additionalSubjectName:
     - {{ headlessSvcName }}
```

Reply:
- yes lets add `additionalSubjectName`, or maybe `additionalSubjectNames` to https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/tls/tls.go#L40 . maybe plural to be able to add a comma separated list of additional SubjectNames?

- Where is the headless service name used/configured? do we have to configure it in the httpd.conf as a `ServerAlias`, iiuc right now we only pass in the svc name https://github.com/openstack-k8s-operators/glance-operator/blob/main/controllers/glanceapi_controller.go#L1015

Reply:
Yes, sorry for being late to this work. I do agree we should have the lib-common part to manage a comma separated list of subject names. They can be built by the service operators, and the openstack-operator can apply them by accessing the annotation.
I'll follow up here in a bit.